controllers/storagecluster/reconcile.go

```diff
@@ -526,6 +531,26 @@ func (obj *ocsCephConfig) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.S
 		return nil
 	}
 
+	// configure public network if the cluster is dualstack, but not multus
+	if sc.Spec.Network != nil && sc.Spec.Network.Provider == "" && sc.Spec.Network.DualStack {
+		log.Info("Prajith DualStack is enabled, but no network provider is specified. Using the default network provider.")
+		//networkConfig := &configv1.Network{}
+		networkConfig := &configv1.Network{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster", Namespace: ""}, networkConfig)
+		if err != nil {
+			return fmt.Errorf("could not get network config details to determine cloud platform: %v", err)
+		}
+		cidrNameArray := []string{}
+		for _, cidr := range networkConfig.Spec.ClusterNetwork {
```

Comment:
The documentation states that the status (not the spec) should not be read. 

https://docs.openshift.com/container-platform/4.8/rest_api/config_apis/network-config-openshift-io-v1.html